### PR TITLE
Feature: Additional options

### PR DESCRIPTION
**Pull request by HugoLaguna**
# Overview

* widget  no have `data-stars`
* widget  no have default values

# Proposed Solution

* props of widget documented
* add additional prop to set `data-stars` on widget
* add default values
* block without badged

# Evidences
image

# Example

`````json
{
  "trustpilot-widget#home":{
    "title":"Widget TrustPilot",
    "props":{
      "width":"100%",
      "height":"240px",
      "theme":"light",
      "stars":"4,5",
      "templateId":"54ad5defcds23c28af8b",
      "businessUnitId":"597a5f232442343243243fe"
    }
  }, 
}
`````

## Workdspace

- https://truspilot--idville.myvtex.com/

## Evidences

image

### CODE REVIEW

**Comment by HugoLaguna** [comment]
Patch (backwards-compatible bug fixes)

**Comment by HugoLaguna** [comment]
Patch (backwards-compatible bug fixes)

**Comment by HugoLaguna** [comment]
@kaisermann  hi! how we can accelerate this PR ? we need this feature to deploy a site and i don't wanna request new app to IO

**Comment by HugoLaguna** [comment]
Minor (backwards-compatible functionality)

**Comment by HugoLaguna** [comment]
@klzns how i can accelerate this PR ? or who is the team to help us on this? we need this props to deploy block on a client and i don't wanna request a new app on IO with the same functionality. you can help mew ? or do you know who can ?

**Comment by klzns** [comment]
@HugoLaguna I no long work at VTEX

I think @victorhmp or @icazevedo can help you with this

**Comment by HugoLaguna** [comment]
@klzns i'm so sorry! , many thanks for response me!

**Review by wender — CHANGES_REQUESTED** [review]
It is missing an addition to the `CHANGELOG.md` file

File: react/Widget.tsx [hunk 1]
```diff
@@ -8,15 +8,17 @@ interface Props {
...
 const Widget: StorefrontFunctionComponent<Props> = ({
   // 5419b6a8b0d04a076446a9ad = micro review count
   templateId = '5419b6a8b0d04a076446a9ad',
-  businessUnitId,
-  theme,
-  width,
-  height,
+  businessUnitId = '',
+  theme = "light",
+  width = "100%",
```
Comment (wender) [inline react/Widget.tsx]: use single quotes instead

File: react/Widget.tsx [hunk 2]
```diff
@@ -8,15 +8,17 @@ interface Props {
...
 
 const Widget: StorefrontFunctionComponent<Props> = ({
   // 5419b6a8b0d04a076446a9ad = micro review count
   templateId = '5419b6a8b0d04a076446a9ad',
-  businessUnitId,
-  theme,
-  width,
-  height,
+  businessUnitId = '',
+  theme = "light",
```
Comment (wender) [inline react/Widget.tsx]: use single quotes instead

File: react/Widget.tsx [hunk 3]
```diff
@@ -8,15 +8,17 @@ interface Props {
...
+  width = "100%",
+  height = "240px",
```
Comment (wender) [inline react/Widget.tsx]: use single quotes instead

File: react/Widget.tsx [hunk 4]
```diff
@@ -8,15 +8,17 @@ interface Props {
...
+  stars = "4,5",
```
Comment (wender) [inline react/Widget.tsx]: use single quotes instead

**Review by wender — CHANGES_REQUESTED** [review]

File: react/Widget.tsx (hunk 1 above)
Comment (HugoLaguna) [reply react/Widget.tsx]: fixed

File: react/Widget.tsx (hunk 3 above)
Comment (HugoLaguna) [reply react/Widget.tsx]: fixed

File: react/Widget.tsx (hunk 4 above)
Comment (HugoLaguna) [reply react/Widget.tsx]: fixed

File: react/Widget.tsx (hunk 2 above)
Comment (HugoLaguna) [reply react/Widget.tsx]: fixed

**Comment by HugoLaguna** [comment]
@wender many thanks for your review!

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+## [2.1.0] - 2021-06-18
+
+### Added
+- Generic trustpilot widget block without badge.
+- Aditional props to widget
+
```
Comment (victorhmp) [inline CHANGELOG.md]: Our release bot will add the date and version numbers when this PR is merged, no need to do it manually :)

```suggestion
## [Unreleased]
### Added
- Generic trustpilot widget block without badge.
- Aditional props to widget

```

File: docs/README.md [hunk 6]
```diff
@@ -38,6 +38,14 @@ The `trustpilot` app also enables Truspilot's widget component to be added to yo
 6. Save your changes;
 7. Access your store's website using the Developer workspace to check out the new component. If no more changes are desired, repeat the steps above in a [Production workspace](https://vtex.io/docs/recipes/development/creating-a-production-workspace/) and then [promote it to Master](https://vtex.io/docs/recipes/development/promoting-a-workspace-to-master).
 
+### Optional
```
Comment (victorhmp) [inline docs/README.md]:
```suggestion
### Optional

```

File: docs/README.md [hunk 7]
```diff
@@ -38,6 +38,14 @@ The `trustpilot` app also enables Truspilot's widget component to be added to yo
...
+In case that recibe error on toolbet `I couldn't find an interface "shop-review-badge.trustpilot-widget"` after finish all steps to install, you can add the Trustpilot app to your theme's manifest.json file inside peerDependencies as shown below:
```
Comment (victorhmp) [inline docs/README.md]: I couldn't quite understand what you meant here, could you please try to rewrite this section?

File: docs/README.md [hunk 8]
```diff
@@ -38,6 +38,14 @@ The `trustpilot` app also enables Truspilot's widget component to be added to yo
...
+`````diff
+ "peerDependencies": {
++   "vtex.trustpilot": "1.x"
+ }
+`````
```
Comment (victorhmp) [inline docs/README.md]: There shouldn't be that many `s, 3 are already enough :)

File: docs/README.md [hunk 9]
```diff
@@ -46,6 +54,31 @@ The `trustpilot` app also enables Truspilot's widgets to be added to your store
 | ------------------- | ------------------- | --------------------------------------- |
 | `shop-review-badge` | `trustpilot-widget` | Adds a Trustpilot widget in your store. |
 
+
+## Custom Blocks props of `trustpilot-widget`
+
+`````json
```
Comment (victorhmp) [inline docs/README.md]: Same think here, just ```json should be enough

File: docs/README.md [hunk 10]
```diff
@@ -46,6 +54,31 @@ The `trustpilot` app also enables Truspilot's widgets to be added to your store
...
+      "width":"100%",
+      "height":"240px",
+      "theme":"light",
+      "stars":"4,5",
+      "businessUnitId":"597a23fwffdsfe05a793fe"
+    }
+  }
+}
+`````
+| Prop name |   Type   |                         Description                         | Default value |
```
Comment (victorhmp) [inline docs/README.md:78]:
```suggestion

| Prop name |   Type   |                         Description                         | Default value |
```

File: manifest.json [hunk 11]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "trustpilot",
   "vendor": "vtex",
-  "version": "2.0.0",
+  "version": "2.1.0-beta.0",
```
Comment (victorhmp) [inline manifest.json]: Updating the version here in the manifest is also handled by a bot. Please revert this back to `2.0.0`

File: messages/pt.json [hunk 12]
```diff
@@ -2,5 +2,9 @@
   "admin/editor.trustpilot.base-url": "URL Base",
   "admin/editor.trustpilot.template-id": "ID do template",
   "admin/editor.trustpilot.business-unit-id": "ID da unidade de negócio",
-  "admin/editor.trustpilot.widget": "Trustpilot - Widget Genérico"
+  "admin/editor.trustpilot.widget": "Trustpilot - Widget Genérico",
+  "admin/editor.trustpilot.theme": "Theme",
+  "admin/editor.trustpilot.width": "Width",
+  "admin/editor.trustpilot.height": "Height",
+  "admin/editor.trustpilot.stars": "Stars"
```
Comment (victorhmp) [inline messages/pt.json]: Shouldn't these be translated to Portuguese too?

File: react/Widget.tsx [hunk 13]
```diff
@@ -8,15 +8,17 @@ interface Props {
...
   templateId = '5419b6a8b0d04a076446a9ad',
-  businessUnitId,
-  theme,
-  width,
-  height,
+  businessUnitId = '',
+  theme = 'light',
+  width = '100%',
+  height = '240px',
+  stars = '4,5',
```
Comment (victorhmp) [inline react/Widget.tsx]: This would change the default values for everyone using the app, is this desirable? Wouldn't it be a breaking change?

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/Widget.tsx (hunk 13 above)
Comment (HugoLaguna) [reply react/Widget.tsx]: it's true. i remove that's not necessary.

File: messages/pt.json (hunk 12 above)
Comment (HugoLaguna) [reply messages/pt.json]: added!

File: manifest.json (hunk 11 above)
Comment (HugoLaguna) [reply manifest.json]: sorry! I thought @Danger was requesting me. i remove them

File: docs/README.md (hunk 10 above)
Comment (HugoLaguna) [reply docs/README.md:78]: fixed

File: docs/README.md (hunk 9 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed

File: docs/README.md (hunk 8 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed

File: docs/README.md (hunk 7 above)
Comment (HugoLaguna) [reply docs/README.md]: I don't know what I meant XD, I didn't understand that that was`shop-review-badge`, i remove this and change for that

```markdown
## Block without badge

To use widget without `shop-review-badge` to landings for example you can use `trustpilot-widget`, you need add to your manifest this....
```

File: docs/README.md (hunk 6 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed

File: CHANGELOG.md (hunk 5 above)
Comment (HugoLaguna) [reply CHANGELOG.md]: fixed

**Comment by HugoLaguna** [comment]
thanks @victorhmp for your review

File: docs/README.md [hunk 14]
```diff
@@ -46,6 +47,43 @@ The `trustpilot` app also enables Truspilot's widgets to be added to your store
...
+}
+```
+
+| Prop name |   Type   |                         Description                         | Default value |
+| :-------: | :------: | :---------------------------------------------------------: | :-----------: |
+|  `width`  | `string` | Change width of template |  `100%`   |
+|  `height`  | `string` | Change height of template |  `240px`   |
+|  `theme`  | `string` | Change theme of template `light` or `dark` |  `light`   |
+|  `stars`  | `string` | Change stars stock of template |  `4,5`   |
+|  `businessUnitId`  | `string` | Set your business code  |  null  |
```
Comment (victorhmp) [inline docs/README.md]: Could you update the default values here to `undefined`?

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 14 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed

Comment (HugoLaguna) [reply docs/README.md]: fixed

**Review by wender — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]
LGTM, but we should have a review in the docs

File: docs/README.md [hunk 15]
```diff
@@ -46,6 +47,43 @@ The `trustpilot` app also enables Truspilot's widgets to be added to your store
 | ------------------- | ------------------- | --------------------------------------- |
 | `shop-review-badge` | `trustpilot-widget` | Adds a Trustpilot widget in your store. |
 
+
+## Block without badge
+
+To use widget without `shop-review-badge` to landings for example you can use `trustpilot-widget`, you need add to your manifest this
```
Comment (mariana-caetano) [inline docs/README.md]:
```suggestion
To use widget without `shop-review-badge` to landing pages, for example, you can use `trustpilot-widget`. Therefore, you need to add the following in the manifest.json file.
```
Hugo, just to clarify, what do you mean by "landings"? Is there any word could we substitute for? Thank you!

File: docs/README.md [hunk 16]
```diff
@@ -38,6 +38,7 @@ The `trustpilot` app also enables Truspilot's widget component to be added to yo
 6. Save your changes;
```
Comment (mariana-caetano) [inline docs/README.md:38]: A good practice when writing steps is to finish each one with a period or a colon as we can check [here](https://www.notion.so/vtexhandbook/Editorial-guidelines-33cb21cab65844d5b35a7a2e1bf87056#b1748ecd144d42619754c2f7492f3261), for example: 
```suggestion
6. Save your changes.
```

File: docs/README.md [hunk 17]
```diff
@@ -38,6 +38,7 @@ The `trustpilot` app also enables Truspilot's widget component to be added to yo
...
 7. Access your store's website using the Developer workspace to check out the new component. If no more changes are desired, repeat the steps above in a [Production workspace](https://vtex.io/docs/recipes/development/creating-a-production-workspace/) and then [promote it to Master](https://vtex.io/docs/recipes/development/promoting-a-workspace-to-master).
 
+
 :information*source: \_The integration key must be provided by Trustpilot*.
```
Comment (mariana-caetano) [inline docs/README.md]:
```suggestion
:information_source: _The integration key must be provided by Trustpilot._
```

**Review by mariana-caetano — CHANGES_REQUESTED** [review]
Awesome, Hugo! I've left some suggestions that should be addressed before it gets merged.

**Review by igorbrasileiro — APPROVED** [review]

**Comment by igorbrasileiro** [comment]
Can you leave a workspace to test?

File: CHANGELOG.md [hunk 18]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- Generic trustpilot widget block without badge.
+- Aditional props to widget
```
Comment (icazevedo) [inline CHANGELOG.md]:
```suggestion
- Additional props to widget.
```

File: docs/README.md (hunk 15 above)
Comment (icazevedo) [reply docs/README.md]: I think he meant landing pages

**Review by icazevedo — APPROVED** [review]

File: docs/README.md (hunk 16 above)
Comment (HugoLaguna) [reply docs/README.md:38]: fixed

File: docs/README.md (hunk 15 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed

File: docs/README.md (hunk 17 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed

**Review by mariana-caetano — APPROVED** [review]

**Comment by HugoLaguna** [comment]
@igorbrasileiro  i updated description of PR with WS and images

File: docs/README.md [hunk 19]
```diff
@@ -35,17 +35,55 @@ The `trustpilot` app also enables Truspilot's widget component to be added to yo
...
+    "props":{
+      "width":"100%",
+      "height":"240px",
+      "theme":"light",
+      "stars":"4,5",
+      "businessUnitId":"597a23fwffdsfe05a793fe",
+      "templateId": "5419b6a8b0d04a076446a9ad",
+    }
+  }
+}
```
Comment (igorbrasileiro) [inline docs/README.md]:
```suggestion
{
   "trustpilot-widget#home":{
      "title":"Widget TrustPilot",
      "props":{
         "width":"100%",
         "height":"240px",
         "theme":"light",
         "stars":"4,5",
         "businessUnitId":"597a23fwffdsfe05a793fe",
         "templateId":"5419b6a8b0d04a076446a9ad"
      }
   }
}
```

File: docs/README.md [hunk 20]
```diff
@@ -35,17 +35,55 @@ The `trustpilot` app also enables Truspilot's widget component to be added to yo
...
 5. When editing the block, add the widget's `template id` and `business unit id` value provided by the [Trustpilot's widget](https://widgets-trustboxinstructions.b2b.trustpilot.com/#/trustbox/eyJjb2RlSGVhZCI6IjwhLS0gVHJ1c3RCb3ggc2NyaXB0IC0tPlxuPHNjcmlwdCB0eXBlPVwidGV4dC9qYXZhc2NyaXB0XCIgc3JjPVwiLy93aWRnZXQudHJ1c3RwaWxvdC5jb20vYm9vdHN0cmFwL3Y1L3RwLndpZGdldC5ib290c3RyYXAubWluLmpzXCIgYXN5bmM%252BPC9zY3JpcHQ%252BXG48IS0tIEVuZCBUcnVzdEJveCBzY3JpcHQgLS0%252BIiwiY29kZUJvZHkiOiI8IS0tIFRydXN0Qm94IHdpZGdldCAtIE1pY3JvIFJldmlldyBDb3VudCAtLT5cbjxkaXYgY2xhc3M9XCJ0cnVzdHBpbG90LXdpZGdldFwiIGRhdGEtbG9jYWxlPVwiaXQtSVRcIiBkYXRhLXRlbXBsYXRlLWlkPVwiNTQxOWI2YThiMGQwNGEwNzY0NDZhOWFkXCIgZGF0YS1idXNpbmVzc3VuaXQtaWQ9XCI1ZTc0ZTJkYjYwMGQxYTAwMDFiZTE1ODlcIiBkYXRhLXN0eWxlLWhlaWdodD1cIjI0cHhcIiBkYXRhLXN0eWxlLXdpZHRoPVwiMTAwJVwiIGRhdGEtdGhlbWU9XCJsaWdodFwiPlxuICA8YSBocmVmPVwiaHR0cHM6Ly9pdC50cnVzdHBpbG90LmNvbS9yZXZpZXcvbWlyaWFkZS5jb21cIiB0YXJnZXQ9XCJfYmxhbmtcIiByZWw9XCJub29wZW5lclwiPlRydXN0cGlsb3Q8L2E%252BXG48L2Rpdj5cbjwhLS0gRW5kIFRydXN0Qm94IHdpZGdldCAtLT4iLCJ0cnVzdEJveE5hbWUiOiJNaWNybyBSZXZpZXcgQ291bnQiLCJsb2NhbGUiOiJpdC1JVCJ9);
-6. Save your changes;
+6. Save your changes.
```
Comment (igorbrasileiro) [inline docs/README.md]:
```suggestion
6. Save your changes;
```

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Two small changes

File: docs/README.md (hunk 19 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed!

File: docs/README.md (hunk 20 above)
Comment (HugoLaguna) [reply docs/README.md]: fixed!

**Comment by HugoLaguna** [comment]
@igorbrasileiro you are the Jedi master of Documentation 👍

**Comment by igorbrasileiro** [comment]
> @igorbrasileiro you are the Jedi master of Documentation 👍

Haha thank you!

**Review by igorbrasileiro — APPROVED** [review]

**Comment by HugoLaguna** [comment]
hi team, what are missing to merge ?

**Pull request merged**